packages/ai-core/src/browser/ai-core-preferences.ts

```diff
@@ -63,26 +63,56 @@ export const aiCorePreferenceSchema: PreferenceSchema = {
         [PREFERENCE_NAME_REQUEST_SETTINGS]: {
             title: nls.localize('theia/ai/core/requestSettings/title', 'Custom Request Settings'),
             markdownDescription: nls.localize('theia/ai/core/requestSettings/mdDescription', 'Allows specifying custom request settings for multiple models.\n\
-            Each object represents the configuration for a specific model. The `modelId` field specifies the model ID, `requestSettings` defines model-specific settings.\n\
-            The `providerId` field is optional and allows you to apply the settings to a specific provider. If not set, the settings will be applied to all providers.\n\
-            Example providerIds: huggingface, openai, ollama, llamafile.\n\
+            Each setting consists of:\n\
+            - `scope`: Defines when the setting applies:\n\
+              - `modelId` (optional): The model ID to match\n\
+              - `providerId` (optional): The provider ID to match (e.g., huggingface, openai, ollama, llamafile)\n\
+              - `agentId` (optional): The agent ID to match\n\
+            - `requestSettings`: Model-specific settings as key-value pairs\n\
+            - `clientSettings`: Client-side message handling settings:\n\
+              - `keepToolCalls` (boolean): Whether to keep tool calls in the context\n\
+              - `keepThinking` (boolean): Whether to keep thinking messages\n\
+            Settings are matched based on specificity (agent: 100, model: 10, provider: 1 points).\n\
             Refer to [our documentation](https://theia-ide.org/docs/user_ai/#custom-request-settings) for more information.'),
             type: 'array',
             items: {
                 type: 'object',
```

Comment:
We should probably add an "additionalProperties: false" here, as the old incompatible settings are not marked with a warning.

Reply:
look at line 104 where the schema is actually defined ;-)

Reply:
We should have an `additionalProperties: false` on the root level of the object. The old settings are not shown as invalid in the editor, e.g. `modelId` and `providerId` are not shown as invalid, although they should be in the `scope` object.